### ISSUE TITLE
Victor Mono: patched version has larger line height than vanilla

### ISSUE THREAD

**Issue by acid-bong** [ryanoasis/nerd-fonts#1317]
#### 🗹 Requirements

- [x] I have searched the [issues](https://github.com/ryanoasis/nerd-fonts/issues) for my issue and found nothing related and/or helpful
- [x] I have searched the [FAQ](https://github.com/ryanoasis/nerd-fonts/wiki/FAQ-and-Troubleshooting) for help
- [x] I have searched the [Wiki](https://github.com/ryanoasis/nerd-fonts/wiki) for help

#### 🎯 Subject of the issue

**Expected:** same line height in vanilla and patched versions.

**Experienced:** cf. the title. No difference between manually patched and Github versions (other than the former looking a bit more torn).

#### 🔧 Your Setup

- _Which font are you using (e.g. `Anonymice Powerline Nerd Font Complete.ttf`)?_
   - _VictorMonoNerdFont-Regular.ttf_ (on the screenshot) and _.otf_
- _Where did you get the file from (download link, self patched, source downloaded from link...)_
   - Github archive and manual patching respectively
- _Which terminal emulator are you using (e.g. `iterm2`, `urxvt`, `gnome`, `konsole`)?_
   - `st` and `alacritty`
- _Are you using OS X, Linux or Windows? And which specific version or distribution?_
   - Gentoo Linux amd64
- _Font settings_
   - Both terminals and the window manager (see the statusbar on the top) have `monospace` as font, I configure Victor Mono globally with `~/.config/fontconfig/fonts.conf`:
```xml
<fontconfig>
...
 <alias>
     <family>monospace</family>
     <prefer><family>VictorMono Nerd Font</family></prefer>
 </alias>
...
</fontconfig>
```

#### ★ Screenshots (Optional)

Screenshots will go in pairs: top is vanilla, bottom is patched, no config edits between the screenshots. Left terminal is Alacritty, right -- st. Note the statusbar size on the top

Font size 10, what i'm using normally. Note that the line height in Alacritty and st in uneven now.
![sqrt-20230713_134209](https://github.com/ryanoasis/nerd-fonts/assets/94849097/376b5772-fcb0-4cda-adfe-b7ac041b0cc7)
![sqrt-20230713_134948](https://github.com/ryanoasis/nerd-fonts/assets/94849097/c1abcc17-0a9a-4fc5-a8b1-9e5b3945ac6f)

Scaled up to about x1.5:
![sqrt-20230713_134153](https://github.com/ryanoasis/nerd-fonts/assets/94849097/a11b6c57-9072-4259-a715-673a4b3b5362)
![sqrt-20230713_134933](https://github.com/ryanoasis/nerd-fonts/assets/94849097/c5fd0350-0a3c-4884-b683-6da8442990f4)

Floating window, still with size 10 in it and 38 rows in height:
![sqrt-20230713_134257](https://github.com/ryanoasis/nerd-fonts/assets/94849097/82d0d648-ef37-4257-90d7-a84060debdb3)
![sqrt-20230713_134638](https://github.com/ryanoasis/nerd-fonts/assets/94849097/2ba84e05-9cc6-497a-8346-7622d85a65f2)

Firefox: no difference visually, but with patched font some text blocks have that scrollbar, assuming there's sorta miscalculation
![изображение](https://github.com/ryanoasis/nerd-fonts/assets/94849097/2004159e-6740-46ca-999e-818c36a9aa1f)
![изображение](https://github.com/ryanoasis/nerd-fonts/assets/94849097/7271e6e7-b566-44ea-82c0-da230ad24045)

**Comment by Finii** [comment]
Thanks for reporting!

Which `Victor Mono` do you call vanilla?
* https://github.com/rubjo/victor-mono
* https://github.com/rubjo/victor-mono-font

See bottom of https://github.com/rubjo/victor-mono/issues/155

**Comment by acid-bong** [comment]
The former, it's installed from Gentoo's GURU repo

**Comment by Finii** [comment]
> Font size 10, what i'm using normally. Note that the line height in Alacritty and st in uneven now.

> Note the statusbar size on the top

Well, the reason for all this is that for example the statusbar is ignoring the gap value that is encoded in the font.
Because this is always just a single line and the gap value does only make sense when there are more lines, that decision might be understandable.
Not so for the terminal emulators. They display the 'vanilla' font **wrong**, because they ignore the gap value.

Nerd Fonts removes the gap on patching, partly because the gap is handled differently by different terminals (as you noticed). And the patched fonts shall be rendered identical on any system - for this reason also the Apple centric HHEA and the Microsoft centric OS2 values are unified in the patched fonts. That also can lead to differences pre-to-post patching on one platform xor the other, but in effect everyone sees the same after patching.

And as I showed you on Gitter, "patched version has larger line height than vanilla" is not true universally. MOST terminal emulators do not ignore the gap value and there unpatched and patched is shown identically.

The reason why Nerd Fonts removes the gap is simple... A gap
* is undified where it is, half top and botton, all top, all bottom are the usual but diverse choices for applications
* but that freedom makes it impossible to place the Powerline glyphs vertically correctly
* People expect Powerline glyphs to fill exactly the line space and that includes the gap on most terminal
* How shall we center the glyphs in that line height if we do not know if the gap is top, bottom, or both halfed?
* Additinally the space of the gap should be empty for all glyphs, but we WANT powerline glyphs to touch from line to line, which contradicts

I hope you see there is no other solution than to remove the gap if you want to support all terminals (on all platforms).

Luckily the newest Victor Mono (i.e. version 1.6.3) does _not have any gaps anymore_ :tada:
But at the moment it is unclear how and when that will become the latest release.

Meanwhile I see no real generic solution to this problem, apart from the code that is already implemented.

If you want to self-patch you could turn the gap re-distribution off, to get a vertically tighter font.

```diff
 def get_btb_metrics(font):
     """ Get the baseline to baseline distance for all three metrics """
     hhea_height = font.hhea_ascent - font.hhea_descent
     typo_height = font.os2_typoascent - font.os2_typodescent
     win_height = font.os2_winascent + font.os2_windescent
-    win_gap = max(0, font.hhea_linegap - win_height + hhea_height)
-    hhea_btb = hhea_height + font.hhea_linegap
-    typo_btb = typo_height + font.os2_typolinegap
+    win_gap = 0
     win_btb = win_height + win_gap
     return (hhea_btb, typo_btb, win_btb, win_gap)
``` 

As you also figured out, applications that do the font rendering correctly (like Firefox) show the vanilla and patched font the same. Also note that Firefox show it less-vertically-compact than your terminal emulators. Correctly-font-handling terminals render the text equal to Firefox.

----

In conclusion, I would just update Victor Mono to 1.562, **but** the Oblique style is not existing in that release. So despite Victor Mono's authors suggestion to take that version it is of course not possible without disappointing ppl who like the Oblique and hate the Italic. :roll_eyes: 

Maybe you have better suggestion?

I guess we have no real answer till this has been solved:

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/64854329-c43b-4480-9bcf-72d26ec7b516)

**Comment by acid-bong** [comment]
Tried with 1.562. It's surprisingly huge (1 size larger than before, but that's a different question), but the line height remains the same with patching. I guess the issue I came with is gone due to a different upstream. Cheers
